packages/dashboard/src/app/views/exploreTemplates/index.js

```diff
@@ -42,6 +44,15 @@ function ExploreTemplates() {
   const [activeTemplate, setActiveTemplate] = useState(null);
   const [activeTemplateIndex, setActiveTemplateIndex] = useState(0);
 
+  const {
+    state: {
+      queryParams: { id: templateIdParam },
+    },
+    actions,
+  } = useRouteHistory();
```

Comment:
nit: tighten up value retrieved from context

```suggestion
  const templateIdParam = useRouteHistory(({state}) => state.queryParams.id);
```